apps/web/src/scenes/TokenDetailPage/TokenDetailAsset.tsx

```diff
@@ -30,33 +36,71 @@ function TokenDetailAsset({ tokenRef, hasExtraPaddingForNote }: Props) {
           }
         }
         media @required(action: THROW) {
+          ... on Media {
+            dimensions {
+              width
+              height
+            }
+          }
           ... on HtmlMedia {
             __typename
           }
         }
 
+        ...useGetPreviewImagesSingleFragment
         ...NftDetailAssetComponentFragment
       }
     `,
     tokenRef
   );
 
   const breakpoint = useBreakpoint();
-  const { aspectRatioType } = useContentState();
 
   const contractAddress = token.contract?.contractAddress?.address ?? '';
   const backgroundColorOverride = getBackgroundColorOverrideForContract(contractAddress);
 
   // We do not want to enforce square aspect ratio for iframes https://github.com/gallery-so/gallery/pull/536
   const isIframe = token.media.__typename === 'HtmlMedia';
   const shouldEnforceSquareAspectRatio =
-    !isIframe &&
-    (aspectRatioType !== 'wide' || breakpoint === size.desktop || breakpoint === size.tablet);
+    !isIframe && (breakpoint === size.desktop || breakpoint === size.tablet);
 
   const { handleNftLoaded } = useNftRetry({
     tokenId: token.dbid,
   });
 
+  const resultDimensions = useMemo(() => {
+    const serverSourcedDimensions = token.media?.dimensions;
+    if (serverSourcedDimensions?.width && serverSourcedDimensions.height) {
+      return fitDimensionsToContainerContain({
+        container: { width: DESKTOP_TOKEN_SIZE, height: DESKTOP_TOKEN_SIZE },
+        source: {
+          width: serverSourcedDimensions.width,
+          height: serverSourcedDimensions.height,
+        },
+      });
+    }
+
+    return {
+      height: DESKTOP_TOKEN_SIZE,
+      width: DESKTOP_TOKEN_SIZE,
+    };
+  }, [token.media?.dimensions]);
```

Comment:
same comment to standardize into a reusable hook

Reply:
replied above